src/config_parser/parser.rs

```diff
@@ -52,4 +52,38 @@ impl Config {
             })
         })
     }
+    /// A helper function which returns an appropriate config file path checking if the config
+    /// file exists on that path.
+    ///
+    /// # Error
+    ///
+    /// Returns a `config file not found!!` error if the config file is not present under following
+    /// paths which are:
+    /// 1. `~/.config/websurfx/` if it not present here then it fallbacks to the next one (2)
+    /// 2. `/etc/xdg/websurfx/config.lua` if it is not present here then it fallbacks to the next
+    ///    one (3).
+    /// 3. `websurfx/` (under project folder ( or codebase in other words)) if it is not present
+    ///    here then it returns an error as mentioned above.
+    fn handle_different_config_file_path() -> Result<String, Box<dyn std::error::Error>> {
+        if Path::new(
+            format!(
+                "{}/.config/websurfx/config.lua",
+                std::env::var("HOME").unwrap()
```

Comment:
Do we need to support deployment on Windows?  Windows might not have
 the `HOME` environment variable?

Reply:
I think for right now we will not provide support for Windows because most of the users of this project will be Linux users because most of the users will be privacy-minded individuals so they will tend to use Linux for the project so for now `No` but in future, if we do get a lot of requests from users then we will consider adding support for `windows` and `macOS`. :slightly_smiling_face: